pipdeptree.py

```diff
@@ -775,9 +781,40 @@ def _get_args():
     return parser.parse_args()
 
 
+def handle_non_host_target(args):
+    of_python = os.path.abspath(args.python)
+    # if target is not current python re-invoke it under the actual host
+    if of_python != os.path.abspath(sys.executable):
+        # there's no way to guarantee that graphviz is available, so refuse
+        if args.output_format:
+            print("graphviz functionality is not supported when querying"
+                  " non-host python", file=sys.stderr)
+            raise SystemExit(1)
+        argv = sys.argv[1:]  # remove current python executable
+        py_at = argv.index('--python')  # plus the new python target
```

Comment:
This wouldn't work if the CLI arg is mentioned with `=` eg. `pipdeptree --python=/some/env/bin/python` which is something that argparse supports. I am thinking if it'd be possible to manipulate the args object and convert it to string at the time of building the command. 

But this is something that can be easily documented for the users, so I wouldn't consider it as a blocker for this PR as such.

Reply:
Can add a slight change to fix it.